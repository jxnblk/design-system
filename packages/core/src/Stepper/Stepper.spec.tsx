import React from 'react'

import { Stepper } from '..'

const children = (
  <React.Fragment>
    <Stepper.Step completed>First Step</Stepper.Step>
    <Stepper.Step active>Second Step</Stepper.Step>
    <Stepper.Step>Third Step</Stepper.Step>
  </React.Fragment>
)

describe('Stepper', () => {
  test('renders without crashing', () => {
    const json = rendererCreateWithTheme(<Stepper />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders with children', () => {
    const json = rendererCreateWithTheme(<Stepper>{children}</Stepper>).toJSON()
    expect(json).toMatchSnapshot()
  })
})
