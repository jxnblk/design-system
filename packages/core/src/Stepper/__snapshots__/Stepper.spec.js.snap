// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stepper renders with children 1`] = `
.c4 {
  color: #007aff;
  font-size: 14px;
}

.c5 {
  color: #007aff;
  font-weight: 700;
  font-size: 14px;
}

.c6 {
  color: #4f6f8f;
  font-size: 14px;
}

.c2 {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  margin-right: 4px;
  width: 16px;
  color: #007aff;
}

.c3 {
  outline: none;
}

.c1 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: inherit;
  font-weight: 600;
  line-height: 1.5;
  cursor: pointer;
  border-radius: 2px;
  border-width: 0;
  border-style: solid;
  font-size: 14px;
  padding: 9.5px 18px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  cursor: default;
  padding: 0;
  font-weight: unset;
}

.c1:disabled {
  cursor: not-allowed;
  color: #4f6f8f;
  background-color: #edf0f3;
}

.c1:hover {
  background-color: #fff;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 >:not(:last-child) {
  margin-right: 32px;
}

<div
  className="c0"
>
  <button
    className="c1"
  >
    <svg
      aria-hidden={false}
      aria-label="Step Completed"
      className="c2 c3"
      color="primary"
      fill="currentcolor"
      focusable={false}
      height={16}
      role="img"
      tabIndex={-1}
      viewBox="0 0 24 24"
      width={16}
    >
      <path
        d="M8.6 15.6l-4.2-4.2L3 12.8l5.6 5.6 12-12L19.2 5 8.6 15.6z"
      />
    </svg>
    <div
      aria-selected={false}
      className="c4"
      color="primary"
      fontSize={1}
    >
      First Step
    </div>
  </button>
  <button
    className="c1"
  >
    <div
      aria-selected={true}
      className="c5"
      color="primary"
      fontSize={1}
    >
      Second Step
    </div>
  </button>
  <button
    className="c1"
  >
    <div
      aria-selected={false}
      className="c6"
      color="text.light"
      fontSize={1}
    >
      Third Step
    </div>
  </button>
</div>
`;

exports[`Stepper renders without crashing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 >:not(:last-child) {
  margin-right: 32px;
}

<div
  className="c0"
/>
`;
